src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -537,6 +541,47 @@ private void DeleteAllWorkloadInstallationRecords()
             }
         }
 
+        string GetWorkloadHistoryDirectory()
+        {
+            return Path.Combine(_workloadMetadataDir, _sdkFeatureBand.ToString(), "history");
```

Comment:
"history" should probably be a constant defined together with `InstalledPacksDir` and similar folder names.